rpcserver.go

```diff
@@ -1892,6 +1904,155 @@ func (r *rpcServer) SendAsset(_ context.Context,
 	}, nil
 }
 
+// BurnAsset burns the given number of units of a given asset by sending them
+// to a provably un-spendable script key. Burning means irrevocably destroying
+// a certain number of assets, reducing the total supply of the asset. Because
+// burning is such a destructive and non-reversible operation, some specific
+// values need to be set in the request to avoid accidental burns.
+func (r *rpcServer) BurnAsset(ctx context.Context,
+	in *taprpc.BurnAssetRequest) (*taprpc.BurnAssetResponse, error) {
+
+	var assetID asset.ID
+	switch {
+	case len(in.GetAssetId()) > 0:
+		copy(assetID[:], in.GetAssetId())
+
+	case len(in.GetAssetIdStr()) > 0:
+		assetIDBytes, err := hex.DecodeString(in.GetAssetIdStr())
+		if err != nil {
+			return nil, fmt.Errorf("error decoding asset ID: %w",
+				err)
+		}
+
+		copy(assetID[:], assetIDBytes)
+
+	default:
+		return nil, fmt.Errorf("asset ID must be specified")
+	}
+
+	if in.AmountToBurn == 0 {
+		return nil, fmt.Errorf("amount to burn must be specified")
+	}
+	if in.ConfirmationText != AssetBurnConfirmationText {
+		return nil, fmt.Errorf("invalid confirmation text, please " +
+			"read API doc and confirm safety measure to avoid " +
+			"accidental asset burns")
+	}
+
+	// We are in a bit of a chicken and egg situation here. We need to know
+	// the first PrevID of the asset we spend to generate the burn script
+	// key, but for that we first need to ask the wallet for available
+	// coins. So we first create a burn key with an _empty_ PrevID (so we
+	// can easily find and replace it later). We don't use the bare NUMS key
+	// here as that is used for asset tombstones, which could collide with
+	// what we're trying to do here.
+	tempBurnKey := asset.NewScriptKey(asset.DeriveBurnKey(asset.PrevID{}))
+
+	internalKey, err := r.cfg.AddrBook.NextInternalKey(
+		ctx, asset.TaprootAssetsKeyFamily,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error deriving internal key: %w", err)
+	}
+
+	tapParams := address.ParamsForChain(r.cfg.ChainParams.Name)
+	vPkt := tappsbt.ForInteractiveSend(
+		assetID, in.AmountToBurn, tempBurnKey, 0, internalKey,
+		&tapParams,
+	)
+
+	// Extract the recipient information from the packet. This basically
+	// assembles the asset ID we want to send to and the sum of all output
+	// amounts.
+	desc, err := tapscript.DescribeRecipients(ctx, vPkt, r.cfg.TapAddrBook)
+	if err != nil {
+		return nil, fmt.Errorf("unable to describe packet recipients: "+
+			"%w", err)
+	}
+
+	fundedVPkt, err := r.cfg.AssetWallet.FundPacket(ctx, desc, vPkt)
+	if err != nil {
+		return nil, fmt.Errorf("error funding address send: %w", err)
+	}
+
+	// If there is only a single output, it means that we are attempting to
+	// burn all the assets that were selected and that there are no passive
+	// assets either. This would mean we'd create a BTC level output with
```

Comment:
Wouldn't this be the same as the tombstone scenario? With the only difference being this new burn NUMs key is used instead of the existing burn/nums key?

Reply:
Or it's even just a send as "normal", with no need for an anchor at all, as it looks sort of like UTXO consolidation.

Reply:
Ah, true. But I guess that way we'd end up sending 1k sats to a BTC output that can't be spent anymore (at least not with the current RPCs). I can remove that restriction if we want and create an issue that we'll eventually need a "sweep BTC from outputs with empty commitments/tombstones/burns".